spec/controllers/callbacks_controller_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+describe CallbacksController, :omniauth do
```

Comment:
I would suggest you to check two additional things here:

1. That if your `request.env` is not populated with proper omniauth mock — then no user is created in the db, and n session is set up.
2. That is a user already exists in the database, session will be created, but no new record will be saved in the DB.

I would suggest you using `context` metaphor of RSpec for this — this way your test cases will be clearly structured.

Reply:
```suggestion
describe CallbacksController, :omniauth do
  describe '#create' do  
    context 'with existing user' do
      let(:user) {create :user}
      before do
        request.env['omniauth.auth']  = mock_auth(user)
        # Is `devise.mapping` actually needed here?
      end
      
      it "doesn't create a new User object" do
        expect { post :github }.to_not change{ User.count }
      end
       
       <etc etc etc>
    end
  
    context 'with new user' do
      let(:user) {build :user}
      before do
        request.env['omniauth.auth']  = mock_auth(user)
      end
      
      it 'creates a new User object' do
        expect { post :github }.to change{ User.count }.by(1)
      end
    end
  
    context 'with invalid payload' do
      before do
        request.env['omniauth.auth'] = { foo: :bar}
      end
    end
  end
```